src/components/SubpageUpload/PageUpload.js

```diff
@@ -0,0 +1,90 @@
+import React from 'react';
+import { makeStyles } from '@material-ui/core/styles';
+import Button from '@material-ui/core/Button';
+import Box from '@material-ui/core/Box';
+import { useState } from 'react';
+import UploadService from "./upload-files.service";
+import Backdrop from '@material-ui/core/Backdrop';
+import CircularProgress from '@material-ui/core/CircularProgress';
+import Snackbar from '@material-ui/core/Snackbar';
+import Alert from '@material-ui/lab/Alert';
+//import * as musicMetadata from 'music-metadata-browser';
+
+const useStyles = makeStyles((theme) => ({
+    root: {
+
+    },
+    input: {
+        display: 'none',
+    },
+    backdrop: {
+        zIndex: theme.zIndex.drawer + 1,
+        color: '#fff',
+    },
+}));
+
+export default function PageUpload() {
+    const classes = useStyles();
+    const [uploadMessage, setUploadMessage] = useState(0);
+    const [uploadProgress, setUploadProgress] = useState(-1);
+    const [backdrop, setBackdrop] = useState(false); //false means enabled
+    const [open, setOpen] = React.useState(false);
+
+    const handleClose = (event, reason) => {
+        if (reason === 'clickaway') {
+            return;
+        }
+        setOpen(false);
+    };
+
+    let upload = (e) => {
+        const currentFile = e.target.files[0];
+        const musicMetadata = require('music-metadata-browser');
+        musicMetadata.parseBlob(currentFile).then(metadata => {
+            console.log(metadata.format.container);
+            if (metadata.format.container === 'MPEG') {
+                if (metadata.format.duration > 0 && metadata.format.duration < 90) {
+                    setBackdrop(true);
+                    setUploadProgress(0);
+                    UploadService.upload(currentFile, (event) => setUploadProgress(Math.round((100 * event.loaded) / event.total)))
+                        .then((response) => { setUploadMessage(response.data.message); setBackdrop(false) }) //response.status
+                        .catch((error) => { setUploadMessage(error.message); setUploadProgress(-1); setBackdrop(false); setOpen(true) })
+                } else {
+                    setUploadMessage("Audio file too long!");
+                    setOpen(true);
+                }
+            } else {
+                setUploadMessage("Only mp3 files allowed!");
+                setOpen(true);
+            }
+        }).catch(e => {
+            setUploadMessage("Only mp3 files allowed!");
+            setOpen(true);
+        });
+    }
+
+    //reset e on fail?
```

Comment:
Yes

Reply:
Not relevant right now, since the function has to be refactored.